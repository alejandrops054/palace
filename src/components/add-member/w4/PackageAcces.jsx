/* Sistema Intelligence 2.0
Módulo Contratos
Versión 1.0
Author: IntregraIT Contratos
Comentarios: xxxxx
*/
import React, { useState} from 'react'
import {Nav, NavItem, NavLink, TabContent, TabPane, Row, Col, Card} from 'reactstrap'
import GrantAccess from '../../packages-access/GrantAccess';
import classnames from 'classnames';
import IncludePackages from '../../packages-access/IncludePackages';



const PackageAcces = () => {

    const [activeTab, setActiveTab] = useState('1');

    const toggle = tab => {
      if(activeTab !== tab) setActiveTab(tab);
    }

    return (
        <div>
      <Nav tabs>
        <NavItem>
          <NavLink
            className={classnames({ active: activeTab === '1' })}
            onClick={() => { toggle('1'); }}
          >
            Grant Access
          </NavLink>
        </NavItem>
        <NavItem>
          <NavLink
            className={classnames({ active: activeTab === '2' })}
            onClick={() => { toggle('2'); }}
          >
            Package Access
          </NavLink>
        </NavItem>
      </Nav>
      <TabContent activeTab={activeTab}>
        <TabPane tabId="1">
             <GrantAccess/>
        </TabPane>
        <TabPane tabId="2">
            <IncludePackages/>
        </TabPane>
      </TabContent>
    </div>
    )
}

export default PackageAcces;
