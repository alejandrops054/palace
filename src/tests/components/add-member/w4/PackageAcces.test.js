/* Sistema Intelligence 2.0
Módulo Contratos
Versión 1.0
Author: IntregraIT Contratos
Comentarios: xxxxx
*/

import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import PackageAcces from '../../../../components/add-member/w4/PackageAcces'

describe('PackageAcces component', () => {
    test('debe mostrar snapshot', () => {

        const wrapper = shallow(<PackageAcces />)
        expect(wrapper).toMatchSnapshot();
    })

})