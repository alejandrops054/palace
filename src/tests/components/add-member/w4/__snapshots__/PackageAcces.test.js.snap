// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageAcces component debe mostrar snapshot 1`] = `
<form
  className="card"
>
  <h3>
    Grant Access
  </h3>
  <div
    className="input-group input-group-sm mb-3"
  >
    <Select
      col={12}
      colIcon={12}
      label="Access hotel"
      option="SS1"
    />
  </div>
  <div
    className="row m-3"
  >
    <div
      className="col-md-2"
    >
      <ToggleCheck
        label="Family category"
        numToggle={1}
      />
    </div>
    <div
      className="col-md-2"
    >
      <ToggleCheck
        label="Governor"
        numToggle={2}
      />
    </div>
    <div
      className="col-md-5"
    >
      <ToggleCheck
        label="Lock Off"
        numToggle={3}
      />
    </div>
  </div>
  <div
    className="row mt-3"
  >
    <TextArea
      className="p-5"
      col={12}
      label="Description"
      placeholder="Standar suite - PR, TG, LBC, JM, LBCB and PC"
    />
    <TextArea
      className="p-2"
      col="12"
      label="Standar suite"
      placeholder=" "
    />
    <TextArea
      className="p-2"
      col="12"
      label="Presidential suite"
      placeholder=" "
    />
  </div>
  <div
    className="col-6 input-group input-group-sm mb-3"
  >
    <Select
      col={9}
      label="Access"
      option="Hotel"
    />
    <ModalPrueba />
  </div>
</form>
`;
